### PR TITLE
Don't show dummy range on crash

### LINKED ISSUE #117: Confusing "unable to open <crate>: File is a directory" error
If `cargo-prusti` panics while verifying a crate, this IDE extension reports an error diagnostics that points to a folder. If the user clicks on that diagnostics, the IDE reports an "unable to open <crate>: File is a directory" error message, which might be confusing.

image
(Thanks to @HKalbasi for raising this issue!)

This behavior is a trade-off. If Prusti crashes, we want to list a position-less error in the list of verification errors, but VSCode's API doesn't really allow that.

### PR DESCRIPTION
**Pull request by HKalbasi**
fix #117

### CODE REVIEW

File: src/diagnostics.ts [hunk 1]
```diff
@@ -610,29 +611,23 @@ export class DiagnosticsManager {
...
-                    dummyRange(),
-                    `Unexpected error: ${errorMessage}. See the log (View -> Output -> Prusti Assistant ...) for more details.`,
-                    vscode.DiagnosticSeverity.Error
-                )
-            });
+            crashed = true;
+            vscode.window.showErrorMessage(`Unexpected error: ${errorMessage}. See the log (View -> Output -> Prusti Assistant ...) for more details.`).then(
+                undefined,
+                err => util.log(`Error: ${err}`)
+            );
```
Comment (fpoli) [inline src/diagnostics.ts:630]:
```suggestion
            util.userError(`Prusti terminated with an unexpected error: ${errorMessage}. See the log (View -> Output -> Prusti Assistant ...) for more details.`);
```
`util.userError` also logs the message, which makes it a bit easier to debug some bug reports.

File: src/diagnostics.ts [hunk 2]
```diff
@@ -641,7 +636,10 @@ export class DiagnosticsManager {
             // Render diagnostics
             this.killAllButton.hide();
             verificationDiagnostics.renderIn(this.target);
-            if (verificationDiagnostics.hasErrors()) {
+            if (crashed) {
+                this.verificationStatus.text = `$(warning) Prusti crashed`;
```
Comment (fpoli) [inline src/diagnostics.ts:640]: What about this message?
```suggestion
                this.verificationStatus.text = `$(error) Verification of ${target} '${escapedFileName}' failed with an unexpected error`;
```

**Comment by fpoli** [comment]
Nice, thanks!

**Comment by fpoli** [comment]
The tests should be fixed by https://github.com/viperproject/prusti-assistant/pull/119

**Comment by fpoli** [comment]
(I had to force-push a fix in #119.)

**Comment by fpoli** [comment]
Thanks! I'll publish a new version next week. I have been asked to pause updates until after a lecture.

**Comment by HKalbasi** [comment]
Thanks for handling this so fast! There is no need to hurry for updating, do it as planned.

**Pull request merged**